sdc/tests/test_dataframe.py

```diff
@@ -1244,43 +1248,62 @@ def test_impl(df):
                 with self.assertRaises(KeyError):
                     sdc_func(df)
 
-    def _test_df_getitem_unicode_idx(self, df, idx):
+    @skip_sdc_jit('DF.getitem unsupported exceptions')
+    def test_df_getitem_unicode_idx_exception_key_error(self):
         def test_impl(df, idx):
             return df[idx]
 
         sdc_func = self.jit(test_impl)
-        pd.testing.assert_series_equal(sdc_func(df, idx), test_impl(df, idx))
+
+        for df in [gen_df(test_global_input_data_float64), pd.DataFrame()]:
+            with self.subTest(df=df):
+                with self.assertRaises(KeyError):
+                    sdc_func(df, 'ABC')
 
     @skip_sdc_jit('DF.getitem unsupported Series name')
-    def test_df_getitem_unicode_idx(self):
+    def test_df_getitem_idx(self):
         dfs = [gen_df(test_global_input_data_float64),
                gen_df(test_global_input_data_float64, with_index=True),
                pd.DataFrame({'A': []})]
         for df in dfs:
             with self.subTest(df=df):
+                self._test_df_getitem_str_literal_idx(df)
                 self._test_df_getitem_unicode_idx(df, 'A')
+                self._test_df_getitem_slice_idx(df)
+                self._test_df_getitem_unbox_slice_idx(df, 1, 3)
 
     @skip_sdc_jit('DF.getitem unsupported Series name')
-    def test_df_getitem_unicode_idx_multiple_types(self):
+    def test_df_getitem_idx_multiple_types(self):
         int_data = [-1, 1, 0]
         float_data = [0.1, 0., -0.1]
         str_data = ['ascii', '12345', '1234567890']
         for a, b, c in permutations([int_data, float_data, str_data], 3):
             df = pd.DataFrame({'A': a, 'B': b, 'C': c})
             with self.subTest(df=df):
+                self._test_df_getitem_str_literal_idx(df)
                 self._test_df_getitem_unicode_idx(df, 'A')
+                self._test_df_getitem_slice_idx(df)
+                self._test_df_getitem_unbox_slice_idx(df, 1, 3)
 
-    @skip_sdc_jit('DF.getitem unsupported exceptions')
-    def test_df_getitem_unicode_idx_exception_key_error(self):
-        def test_impl(df, idx):
-            return df[idx]
+    @unittest.skip('DF.getitem unsupported integer columns')
+    def test_df_getitem_int_literal_idx(self):
+        def test_impl(df):
+            return df[1]
 
         sdc_func = self.jit(test_impl)
+        df = gen_df_int_cols(test_global_input_data_float64)
 
-        for df in [gen_df(test_global_input_data_float64), pd.DataFrame()]:
-            with self.subTest(df=df):
-                with self.assertRaises(KeyError):
-                    sdc_func(df, 'ABC')
+        pd.testing.assert_series_equal(sdc_func(df), test_impl(df))
+
+    @unittest.skip('DF.getitem unsupported idx as a tuple')
```

Comment:
Are you planning to add this functionality later?

Reply:
Now I'm not able to get values of a tuple as a literal. So I cannot get data of DF via `get_frame_data`. I will support it when will be able.

Reply:
I gonna support it in separate PR, at the least try to do that.